platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/model/TaskNameComparator.java

```diff
@@ -44,13 +46,7 @@ private int compareSegments(String taskName1, String taskName2) {
         return colon1 == -1 ? taskName1.compareTo(taskName2) : compareSegments(taskName1.substring(colon1 + 1), taskName2.substring(colon2 + 1));
     }
 
-    private int getDepth(String taskName) {
-        int counter = 0;
-        for (char c : taskName.toCharArray()) {
-            if (c == ':') {
-                counter++;
-            }
-        }
-        return counter;
+    static int getDepth(String taskName) {
+        return StringUtils.countMatches(taskName, ":");
```

Comment:
💭  Unrelated to the current changes, but it's a bit of a code smell that we check the depth based on the string content. Ideally, we should work with something like a `Path.segmentCount()` abstraction, but even then included builds might be weird, because they might bring more than one "semicolon".

Just as food for thought, maybe the code using this comparator can be refactored